Initial implementation (#1)
projectsyn/component-metallb: docs/modules/ROOT/pages/index.adoc

= metallb: A Commodore component to manage metallb

{doctitle} is a Commodore component for Managing metallb. The component is based on the https://hub.kubeapps.com/charts/bitnami/metallb[Helm Chart] of Bitnami, which is https://github.com/metallb/metallb/issues/653[suggested] to become a template for the official Helm Chart of the project. Currently there is no support of taking over an existing installation executed by the manifests or kustomize.

See the xref:references/parameters.adoc[parameters] reference for further details.

== Layer2 Mode (default)

For a minimal configuration providing a VIP is required. For specific distributions i.e. Rancher, the loadbalancer ingress-nginx object is created automatically.

```
parameters:
  metallb:
    addresses:
      - 10.128.1.200/32
```

As speaker secretkey must be generated and stored inside the secret key management. A pregenerated key ensures, manifests applied multiple times, the secretkey does not change unexpectedly lead to failovers, because the speakers can not communicate anymore during a rollover unsing different secretkeys.

```

> vault kv put -cas=0 kv/${TENANT_ID}/${CLUSTER_ID}/metallb-speaker secretkey="$(openssl rand -base64 128)"

Author: The value in the secret on the cluster did match the value in Vault. So yes.

= MetalLB: A Commodore component to manage MetalLB

{doctitle} is a Commodore component for Managing MetalLB.
The component is based on the https://hub.kubeapps.com/charts/bitnami/metallb[Helm Chart] of Bitnami.
The Bitnami Helm Chart is https://github.com/metallb/metallb/issues/653[suggested] to become a template for the official Helm Chart of the project.

[CAUTION]
====

> Currently there is no support of taking over an existing installation executed by the manifests or Kustomize.

Reviewer: ```suggestion
Currently there is no support for taking over an existing installation based on the https://metallb.universe.tf/installation/[official guide]
```

====

See the xref:references/parameters.adoc[parameters] reference for further details.

== Layer2 Mode (default)

> For a minimal configuration providing a VIP is required.

Reviewer: ```suggestion
For a minimal configuration, a CIDR defining the VIP range is required.
```

```
parameters:
  metallb:
    addresses:
      - 10.128.1.200/32
```

As memberlist key must be generated and stored inside the secret key management.

> A pregenerated key ensures, manifests applied multiple times, the secret key does not change unexpectedly lead to failovers, because the speakers can not communicate anymore during a rollover using different secret keys.

Reviewer: ```suggestion
A pregenerated key ensures that the manifests can be applied multiple times without the secret key changing.
Otherwise, elevated fail-over rates may occur during manifest application, as the speakers can not communicate during the rollout, due to differing keys.
```

```
vault kv put -cas=0 clusters/kv/${TENANT_ID}/${CLUSTER_ID}/metallb-memberlist secretkey="$(openssl rand -base64 128)"
```

== BGP Mode

> This component currently does not allow to configure MetalLB in the BGP mode. 

Reviewer: This isn't true anymore, by passing `metallb.config` users are free to configure BGP mode for one or more pools

= MetalLB: A Commodore component to manage MetalLB

{doctitle} is a Commodore component for Managing MetalLB.
The component is based on the https://hub.kubeapps.com/charts/bitnami/metallb[Helm Chart] of Bitnami.
The Bitnami Helm Chart is https://github.com/metallb/metallb/issues/653[suggested] to become a template for the official Helm Chart of the project.

[CAUTION]
====
Currently there is no support for taking over an existing installation based on the https://metallb.universe.tf/installation/[official guide]
====

See the xref:references/parameters.adoc[parameters] reference for further details.

== Prerequisites

> As memberlist key must be generated and stored inside the secret key management.

Reviewer 2: ```suggestion
A memberlist key must be generated and stored inside the secret key management.
```
